code-dot-org/code-dot-org
# write exact-match ai scores to the database in eval rubric job

Depends on https://github.com/code-dot-org/code-dot-org/pull/57066. Continues https://codedotorg.atlassian.net/browse/AITT-489. This PR does the folllowing when the eval rubric job is run:
* read confidence-exact.json
* use learning goal and ai label to figure out which confidence value to pull out of the json
* write the value to the database

Also done behind the scenes to prop up this PR:
* [spreadsheet](https://docs.google.com/spreadsheets/d/1q852ucSF_BsKkKjozxTumLqq0_XORe0IliU6De3YDIU/edit#gid=0) summarizing which [lesson, learning goal, ai score] tuples are high confidence, following the criteria in the [Accuracy Thresholds doc](https://docs.google.com/document/d/1OCtVDskN0z4-_vYBgt3Bv7P9xB6IlIn5b3mEEeCR0nY/edit)
* created a new release `2024-03-06-ai-rubrics-json-reason-confidence-exact/` and uploaded it to S3

## Testing story

manual testing:
* temporarily pointed local dashboard to `2024-03-06-ai-rubrics-json-reason-confidence-exact/`, and manually verified that the correct values are getting written to the DB
* also used a work-in-progress UI PR to confirm that the ai_confidence_exact_match values are getting passed through to the UI and that the correct values are being set

unit test updates:
* updated s3 stub to provide fake confidence-exact.json
* updated all existing unit tests to verify that valid `ai_confidence_exact_match` values are written to the DB
* added a unit test covering the case where confidence-exact.json is not present, in which case `nil` values should be written to the db

## Future work

* launch confidence-exact.json files as part of a release. currently I am planning to wait for @wilkie's addition of evidence prompting to other lessons before doing this
